Tests/Shared.Specs/ExceptionAssertionSpecs.cs

```diff
@@ -895,6 +894,62 @@ public void When_no_exception_should_be_thrown_and_none_was_it_should_not_throw(
             //-----------------------------------------------------------------------------------------------------------
             foo.Invoking(f => f.Do()).Should().NotThrow();
         }
+
+        private static void ThrowUntil(Stopwatch watch, int throwTime)
+        {
+                if(watch.ElapsedMilliseconds <= throwTime)
+                    throw new ArgumentException("An exception was forced");
+        }
+
+        [Fact]
+        public void When_no_exception_should_be_thrown_after_wait_time_but_it_was_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var foo = A.Fake<IFoo>();
+            var watch = Stopwatch.StartNew();
+            var waitTime = 100;
+            var pollInterval = 10;
+
+            void ThrowLongerThanWaitTime() => ThrowUntil(watch, waitTime + waitTime / 2);
+
+            A.CallTo(() => foo.Do()).Invokes(ThrowLongerThanWaitTime);
```

Comment:
🙃 Can this be simplified to the following?

```csharp
Action throwLongerThanWaitTime = () => ThrowUntil(watch, waitTime + waitTime / 2);

//-----------------------------------------------------------------------------------------------------------
// Act
//-----------------------------------------------------------------------------------------------------------
Action act = () => throwLongerThanWaitTime.Should().NotThrowAfter(waitTime, pollInterval, "we passed valid arguments");

//-----------------------------------------------------------------------------------------------------------
// Assert
//-----------------------------------------------------------------------------------------------------------
act.Should().Throw<XunitException>()
    .WithMessage("*we passed valid arguments*");
```

Reply:
> 🙃 Can this be simplified to the following?
Sure, but the same would apply to some of the surrounding tests? I also do not really get why the mocking should be used here. @dennisdoomen Should the Arrange-Act-Assert structure of the other tests by strictly imitated?

Reply:
You should always divide the tests into comment separated Arrange, Act & Assert sections.
There are a few exceptions to this, where e.g. Arrange and Act sections are collapsed into one section.

If existing tests has room for improvements, you do not need to copy their bad behavior.
If you have specific tests in mind, you can write me and I'll include them in occasional cleanup PRs.

FYI: make a blank line after quoting someone to separate your comment in Github Markdown.